Mathlib/Logic/Equiv/TransferInstance.lean

```diff
@@ -43,131 +43,132 @@ section Instances
   rfl
 #align equiv.div_def Equiv.div_def
 #align equiv.sub_def Equiv.sub_def
 
-/-- Transfer `has_inv` across an `equiv` -/
-@[reducible, to_additive "Transfer `has_neg` across an `equiv`"]
-protected def hasInv [Inv β] : Inv α :=
+/-- Transfer `Inv` across an `Equiv` -/
+@[reducible, to_additive "Transfer `Neg` across an `Equiv`"]
+protected def Inv [Inv β] : Inv α :=
   ⟨fun x => e.symm (e x)⁻¹⟩
-#align equiv.has_inv Equiv.hasInv
-#align equiv.has_neg Equiv.hasNeg
+#align equiv.has_inv Equiv.Inv
+#align equiv.has_neg Equiv.Neg
 
 @[to_additive]
-theorem inv_def [Inv β] (x : α) : @Inv.inv _ (Equiv.hasInv e) x = e.symm (e x)⁻¹ :=
+theorem inv_def [Inv β] (x : α) : @Inv.inv _ (Equiv.Inv e) x = e.symm (e x)⁻¹ :=
   rfl
 #align equiv.inv_def Equiv.inv_def
 #align equiv.neg_def Equiv.neg_def
 
-/-- Transfer `has_smul` across an `equiv` -/
+/-- Transfer `SMul` across an `Equiv` -/
 @[reducible]
-protected def hasSmul (R : Type _) [SMul R β] : SMul R α :=
+protected def SMul (R : Type _) [SMul R β] : SMul R α :=
   ⟨fun r x => e.symm (r • e x)⟩
-#align equiv.has_smul Equiv.hasSmul
+#align equiv.has_smul Equiv.SMul
 
 theorem smul_def {R : Type _} [SMul R β] (r : R) (x : α) :
     @SMul.smul _ _ (e.SMul R) r x = e.symm (r • e x) :=
   rfl
 #align equiv.smul_def Equiv.smul_def
 
-/-- Transfer `has_pow` across an `equiv` -/
-@[reducible, to_additive SMul]
-protected def hasPow (N : Type _) [Pow β N] : Pow α N :=
+/-- Transfer `Pow` across an `Equiv` -/
+@[reducible, to_additive existing SMul]
+protected def Pow (N : Type _) [Pow β N] : Pow α N :=
   ⟨fun x n => e.symm (e x ^ n)⟩
-#align equiv.has_pow Equiv.hasPow
-#align equiv.has_smul Equiv.hasSmul
+#align equiv.has_pow Equiv.Pow
 
 theorem pow_def {N : Type _} [Pow β N] (n : N) (x : α) :
     @Pow.pow _ _ (e.Pow N) x n = e.symm (e x ^ n) :=
   rfl
 #align equiv.pow_def Equiv.pow_def
 
-/-- An equivalence `e : α ≃ β` gives a multiplicative equivalence `α ≃* β`
-where the multiplicative structure on `α` is
-the one obtained by transporting a multiplicative structure on `β` back along `e`.
--/
-@[to_additive
-      "An equivalence `e : α ≃ β` gives a additive equivalence `α ≃+ β`\nwhere the additive structure on `α` is\nthe one obtained by transporting an additive structure on `β` back along `e`."]
+/-- An equivalence `e : α ≃ β` gives a multiplicative equivalence `α ≃* β` where
+the multiplicative structure on `α` is the one obtained by transporting a multiplicative structure
+on `β` back along `e`. -/
+@[to_additive "An equivalence `e : α ≃ β` gives an additive equivalence `α ≃+ β` where
+the additive structure on `α` is the one obtained by transporting an additive structure
+on `β` back along `e`."]
 def mulEquiv (e : α ≃ β) [Mul β] :
-    letI := Equiv.hasMul e
+    let mul := Equiv.Mul e
     α ≃* β := by
   intros
   exact
     { e with
       map_mul' := fun x y => by
         apply e.symm.injective
-        simp }
+        simp only [toFun_as_coe_apply, symm_apply_apply]
+        rfl }
```

Comment:
It would be better to try and fix `simp` than to insert these workarounds.

Reply:
It's not much of a problem – just `simp; rfl` also works. Should I do that?

Reply:
What you should do is look at the goal state you end up with, and try to work out which simp lemma is missing.

Reply:
The culprit was `add/mul_def`...